src/main/java/com/ays/backend/user/security/JwtTokenProvider.java

```diff
@@ -0,0 +1,68 @@
+package com.ays.backend.user.security;
+
+
+import io.jsonwebtoken.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+public class JwtTokenProvider {
+
+    @Value("${afet.app.secret}")
+    private String APP_SECRET;
+
+    @Value("${afet.expires.in}")
+    private long EXPIRES_IN;
+
+    public String generateJwtToken(Authentication auth) {
+        UserDetails userDetails = (UserDetails) auth.getPrincipal();
+        Date now = new Date();
+        Date expiryDate = new Date(now.getTime() + EXPIRES_IN);
+
+        String token = Jwts.builder()
+                .setSubject(userDetails.getUsername())
+                .setIssuedAt(now)
+                .setExpiration(expiryDate)
+                .signWith(SignatureAlgorithm.HS512, APP_SECRET)
+                .compact();
+
+        return token;
+    }
+
+    public String getUserNameFromJwtToken(String token) {
+        return Jwts.parser().setSigningKey(APP_SECRET).parseClaimsJws(token).getBody().getSubject();
+    }
+
+    public Long getUserIdFromJwt(String token) {
+        Claims claims = Jwts.parser()
+                .setSigningKey(APP_SECRET)
+                .parseClaimsJws(token)
+                .getBody();
+
+        return Long.parseLong(claims.getSubject());
+    }
+
+    boolean validateToken(String token) {
+        try {
+            Jwts.parser().setSigningKey(APP_SECRET).parseClaimsJws(token);
+            return !isTokenExpired(token);
+        } catch (MalformedJwtException e) {
+            return false;
+        } catch (ExpiredJwtException e) {
+            return false;
+        } catch (UnsupportedJwtException e) {
+            return false;
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
```

Comment:
We can write the catch block as follows to make it more readable.

```
catch (MalformedJwtException | ExpiredJwtException | UnsupportedJwtException | I llegalArgumentException exception) {
   return false;
}
```

Reply:
I just revised the code snippet as you said.